src/ripple/app/transactors/test/MultiSign.test.cpp

```diff
@@ -0,0 +1,1611 @@
+            aliceBalanceCheck (0);
+        }
+        // Set a signer list where the quorum can't be met.  Should fail.
+        {
+            SignerList aliceSigners {{bogie, 3}, {ghost, 3}, {haunt, 3}};
+            STTx tx = getSignerListSetTx (alice, aliceSigners, 10);
+            singleSign (tx, alice);
+            ledger.applyBadTransaction (tx, temBAD_QUORUM);
+            alice.decrSeq (); // Fix up local account sequence number.
+            aliceBalanceCheck (0);
+        }
+        // Try setting a signer list where the quorum can barely be met.  Also,
+        // set a weight of zero, which is legal.
+        {
+            SignerList aliceSigners{{bogie, 0}, {ghost, 65535}, {haunt, 65535}};
+            STTx tx = getSignerListSetTx (alice, aliceSigners, 131070);
+            singleSign (tx, alice);
+            ledger.applyGoodTransaction (tx);
+            aliceBalanceCheck (stdFee);
+        }
+        // Try a zero quorum.  Should fail.
+        {
+            SignerList aliceSigners{{bogie, 0}, {ghost, 0}, {haunt, 0}};
+            STTx tx = getSignerListSetTx (alice, aliceSigners, 0);
+            singleSign (tx, alice);
+            ledger.applyBadTransaction (tx, temMALFORMED);
+            alice.decrSeq (); // Fix up local account sequence number.
+            aliceBalanceCheck (0);
+        }
+
+        // Try to create a signer list that's barely too big.  Should fail.
+        UserAccount becky (algorithm, "becky");
+        payInDrops (ledger, master, becky, 1000*xrp);
+
+        UserAccount cheri (algorithm, "cheri");
+        payInDrops (ledger, master, cheri, 1000*xrp);
+
+        UserAccount daria (algorithm, "daria");
+        payInDrops (ledger, master, daria, 1000*xrp);
+        {
+            SignerList aliceSigners{{bogie, 1}, {ghost, 1}, {haunt, 1},
+                {jinni, 1}, {shade, 1}, {spook, 1}, {becky, 1}, {cheri, 1},
+                {daria, 1}};
+            STTx tx = getSignerListSetTx (alice, aliceSigners, 1);
+            singleSign (tx, alice);
+            ledger.applyBadTransaction (tx, temMALFORMED);
+            alice.decrSeq (); // Fix up local account sequence number.
+            aliceBalanceCheck (0);
+        }
+        // Make the biggest allowed list.  This one should succeed.
+        {
+            SignerList aliceSigners{{bogie, 1}, {ghost, 1}, {haunt, 1},
+                {jinni, 1}, {shade, 1}, {spook, 1}, {becky, 1}, {cheri, 1}};
+            STTx tx = getSignerListSetTx (alice, aliceSigners, 1);
+            singleSign (tx, alice);
+            ledger.applyGoodTransaction (tx);
+            aliceBalanceCheck (stdFee);
+        }
+        // Remove alice's SignerList.  Should succeed.
+        {
+            SignerList aliceSigners;
+            STTx tx = getSignerListSetTx (alice, aliceSigners, 0);
+            singleSign (tx, alice);
+            ledger.applyGoodTransaction (tx);
+             aliceBalanceCheck (stdFee);
```

Comment:
Great idea.  I'm adding the OwnerCount check right now.  Are there other "things like" OwnerCount I should be checking as well?